allure-testng-adaptor/src/test/java/ru/yandex/qatools/allure/testng/AllureTestListenerConfigMethodsTest.java

```diff
@@ -0,0 +1,127 @@
+package ru.yandex.qatools.allure.testng;
+
+import org.junit.*;
+import org.testng.TestNG;
+import org.testng.xml.XmlClass;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+
+import ru.yandex.qatools.allure.model.ObjectFactory;
+import ru.yandex.qatools.allure.model.Status;
+import ru.yandex.qatools.allure.model.TestCaseResult;
+import ru.yandex.qatools.allure.model.TestSuiteResult;
+import ru.yandex.qatools.allure.utils.AllureResultsUtils;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+
+import java.io.*;
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static java.nio.file.FileVisitResult.*;
+import static ru.yandex.qatools.allure.commons.AllureFileUtils.listTestSuiteFiles;
+
+public class AllureTestListenerConfigMethodsTest {
+
+    private static final String ALLURE_RESULTS = "allure-results";
+
+    private static Path resultsDir;
+    
+    private static final String SUITE_PREFIX = "SUITE"; 
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        resultsDir = Files.createTempDirectory(ALLURE_RESULTS);
+        AllureResultsUtils.setResultsDirectory(resultsDir.toFile());
+        String[] configMethods = {
+            "BeforeSuite","BeforeTest","BeforeClass","BeforeGroups","BeforeMethod",
+            "AfterSuite","AfterTest","AfterClass","AfterGroups","AfterMethod"
+        };
+        List<XmlSuite> suites = new ArrayList<>();
+        for (String configMethod : configMethods) {
+            suites.add(createSuite(configMethod));
+        }
+        TestNG testNG = new TestNG();
+        testNG.setXmlSuites(suites);
+        testNG.setUseDefaultListeners(false);
+        testNG.run();
+    }
+
+    private static XmlSuite createSuite(String testName) {
+        XmlSuite suite = new XmlSuite();
+        suite.setName(SUITE_PREFIX + testName);       
+        XmlTest test = new XmlTest(suite);
+        test.setName(testName);
+        List<XmlClass> classes = new ArrayList<>();
+        classes.add(new XmlClass("ru.yandex.qatools.allure.testng.testdata.ConfigMethodsTest"));
+        test.setXmlClasses(classes);
+        return suite;
+    }
+    
+    
+    @AfterClass
+    public static void tearDown() throws IOException {
+        AllureResultsUtils.setResultsDirectory(null);
+        deleteNotEmptyDirectory(resultsDir);
+    }
+
+  
+    @Test
+    public void validatePendingTest() throws JAXBException {
+        JAXBContext context = JAXBContext.newInstance(ObjectFactory.class);
```

Comment:
You can use this method instead https://github.com/allure-framework/allure-core/blob/master/allure-commons/src/main/java/ru/yandex/qatools/allure/commons/AllureFileUtils.java#L68